lib/ansible/modules/cloud/azure/azure_rm_managed_disk.py

```diff
@@ -385,6 +402,9 @@ def is_different(self, found_disk, new_disk):
         if new_disk.get('tags') is not None:
             if not found_disk['tags'] == new_disk['tags']:
                 resp = True
+        if new_disk.get('zones') is not None:
+            if not found_disk['zones'] == new_disk['zones']:
```

Comment:
The found_disk['zones'] can be `str('')` while it the same as new_disk as None. I would change the `to_dict()` method as another comment.